core/appeal/service_test.go

```diff
@@ -1577,6 +1577,793 @@ func (s *ServiceTestSuite) TestCreate() {
+							Name:        "test",
+							Description: "test",
+							Type:        "http",
+							Config: map[string]interface{}{
+								"url":    url,
+								"method": "GET",
+							},
+							Value: "$response.body.message",
+						},
+					},
+				},
+			},
+		}
+		expectedExistingAppeals := []*domain.Appeal{}
+		expectedActiveGrants := []domain.Grant{
+			{
+				ID:         "99",
+				AccountID:  accountID,
+				ResourceID: "1",
+				Resource: &domain.Resource{
+					ID:  "1",
+					URN: "urn",
+				},
+				Role:           "role_id",
+				Status:         domain.GrantStatusActive,
+				ExpirationDate: &expDate,
+			},
+		}
+		expectedResourceFilters := domain.ListResourcesFilter{IDs: []string{resources[0].ID}}
+		expectedExistingAppealsFilters := &domain.ListAppealsFilter{
+			Statuses:   []string{domain.AppealStatusPending},
+			AccountIDs: []string{"addOnBehalfApprovedNotification-user"},
+		}
+
+		appeals := []*domain.Appeal{
+			{
+				CreatedBy:  accountID,
+				AccountID:  "addOnBehalfApprovedNotification-user",
+				ResourceID: "1",
+				Resource: &domain.Resource{
+					ID:  "1",
+					URN: "urn",
+				},
+				Role:        "role_id",
+				Description: "The answer is 42",
+			},
+		}
+
+		defer server.Close()
+
+		h.mockResourceService.EXPECT().
+			Find(mock.Anything, expectedResourceFilters).Return(resources, nil).Once()
+		h.mockProviderService.EXPECT().
+			Find(mock.Anything).Return(providers, nil).Once()
+		h.mockPolicyService.EXPECT().
+			Find(mock.Anything).Return(policies, nil).Once()
+		h.mockRepository.EXPECT().
+			Find(h.ctxMatcher, expectedExistingAppealsFilters).
+			Return(expectedExistingAppeals, nil).Once()
+		for _, a := range appeals {
+			h.mockGrantService.EXPECT().
+				List(h.ctxMatcher, domain.ListGrantsFilter{
+					Statuses:    []string{string(domain.GrantStatusActive)},
+					AccountIDs:  []string{a.AccountID},
+					ResourceIDs: []string{a.ResourceID},
+					Roles:       []string{a.Role},
+					OrderBy:     []string{"updated_at:desc"},
+				}).
+				Return(expectedActiveGrants, nil).Once()
+		}
+		h.mockProviderService.EXPECT().
+			ValidateAppeal(mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
+		h.mockProviderService.EXPECT().
+			GetPermissions(mock.Anything, mock.Anything, mock.AnythingOfType("string"), "role_id").
+			Return([]interface{}{"test-permission-1"}, nil)
+
+		actualError := h.service.Create(context.Background(), appeals)
+		s.NotNil(actualError)
+		time.Sleep(time.Millisecond)
+		h.assertExpectations(s.T())
+	})
+
+	s.Run("should return appeals on success with metadata sources for post method", func() {
+		h := newServiceTestHelper()
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Here you can specify what the server should return when it receives a request
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte(`{"message": "success"}`))
+		}))
+
+		body := "'{\"user\": ' + $appeal.account_id + '}'"
```

Comment:
Where are we validating that this expression is evaluated properly in this TC?

Reply:
Okay, understood your point. I have added a condition when calling the mock http server which will respond with StatusOk only in case when the body is as expected else it will be error. This way we can validate that correct expression was evaluated.